qiskit/converters/dag_to_circuit.py

```diff
@@ -66,7 +66,7 @@ def dag_to_circuit(dag, copy_operations=True):
     )
     for var in dag.iter_declared_vars():
         circuit.add_uninitialized_var(var)
-    circuit.metadata = dag.metadata
+    circuit.metadata = dict(dag.metadata)
     circuit.calibrations = dag.calibrations
```

Comment:
IIRC, I did this to force a dict input because this was originally storing `.metadata` as a `PyDict` and something was violating that typing. In the python dag this didn't matter because but with `PyDict` it became a type error. But in intervening commits I switched metadata to be a `PyObject` so we can remove this now.

Reply:
Oh actually I got this backwards I thought this was `circuit_to_dag` not `dag_to_circuit`, TBH I'm not sure why I added this here it was probably like 3 weeks ago. I'll see what happens when I remove it now.

Reply:
Removed in: https://github.com/Qiskit/qiskit/pull/12550/commits/d68743f18c2ca6e90fc27b07773a45f516a4fe2d